tests/security/audit/autrace.pm

```diff
@@ -0,0 +1,59 @@
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-Later
+#
+# Summary: Verify the "autrace" utility traces individual processes in a fashion similar to strace. The output of autrace is logged to the audit log.
+# Maintainer: llzhao <llzhao@suse.com>, shawnhao <weixuan.hao@suse.com>
+# Tags: poo#81772
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub get_pid {
+    # Call this fuction to extract pid from autrace output in command line
+    script_run("tail -1 /tmp/out > /tmp/backup");
+    script_run("awk -F\\' '{print \$2}' /tmp/backup > /tmp/out");
+    my $pid = script_output('cat /tmp/out | cut -d " " -f 4');
+    return ($pid);
+}
+
+sub run {
+    my $audit_log = '/var/log/audit/audit.log';
+
+    select_console 'root-console';
+
+    # Install related packages and start audit service
+    zypper_call('in audit libaudit1');
+    assert_script_run("systemctl start auditd");
+
+    # Use autrace to trace an individual process, output will be logged to audit log
+    script_run('autrace /bin/ls /tmp');
+
+    # Delete all existing rules
+    assert_script_run('auditctl -D');
+
+    # Clear original audit log
+    assert_script_run("echo '' > $audit_log");
+
+    # Trace process again and record the pid
+    assert_script_run('autrace /bin/ls > /tmp/out');
+    my $pid = get_pid();
+
+    # Run ausearch with the pid
+    assert_script_run("ausearch -i -p $pid");
+
+    # Clear log again
+    assert_script_run("echo '' > $audit_log");
+
+    # Try resource usage mode and record the pid
+    assert_script_run("autrace -r /bin/ls > /tmp/out");
+    $pid = get_pid();
+
+    # Query audit daemon logs by pid and generate audit reports
+    assert_script_run("ausearch --start recent -p $pid --raw | aureport --file --summary");
+    assert_script_run("ausearch --start recent -p $pid --raw | aureport --host --summary");
```

Comment:
If "--start recent" shows nothing we can try the following parameters:
```
You may also use the word: now, recent, boot, today, yesterday, this-week, week-ago, this-month, this-year, or checkpoint. Boot means the time of  day  to
              the  second  when  the system last booted. Today means starting at 1 second after midnight. Recent is 10 minutes ago. Yesterday is 1 second after midnight
              the previous day. This-week means starting 1 second after midnight on day 0 of the week determined by your locale (see localtime). Week-ago means starting
              1 second after midnight exactly 7 days ago. This-month means 1 second after midnight on day 1 of the month. This-year means the 1 second after midnight on
              the first day of the first month.
```

Reply:
I just tried these parameters and none of them is give me outputs. I guess it is related to '--host' parameter rather than 'recent'. Cuz '--file' is generating outputs while '--host' isn't.

Reply:
Sure.